### ISSUE TITLE
Build error with npm 3.4.1

### ISSUE THREAD

**Issue by conraid** [atom/atom#9591]
Building Atom 1.2.0 and 1.2.1 with NPM 3.4.1 there is this error:

```
Node: v5.0.0
npm: v3.4.1
Installing build modules...
WARN engine prettyjson@0.8.1: wanted: {"node":">= 0.6.0 < 0.11.0"} (current: {"node":"5.0.0","npm":"3.4.1"})
Installing apm...
module.js:339
    throw err;
    ^

Error: Cannot find module 'npm-registry-client'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/tmp/pkg/atom-1.2.1/build/node_modules/npm/lib/cache/caching-client.js:9:22)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
```

with NPM 3.4.0 it's ok

I use LInux (Slackware 64 current)

**Comment by thealjey** [comment]
@conraid the issue seems to be in npm itself [#10394](https://github.com/npm/npm/issues/10394)

**Comment by conraid** [comment]
With npm 3.5.0 it's ok.

**Comment by steezeburger** [comment]
I was able to get everything to build and install properly using nvm with node v4.2.0 and npm v2.14.7

**Comment by InsidiousMind** [comment]
@conraid do you mean npm 3.4.0?

**Comment by middle-aged-man** [comment]
Same issue on Arch LInux..

==> Starting build()...
Node: v5.1.0
npm: v3.4.1
Installing build modules...
WARN engine prettyjson@0.8.1: wanted: {"node":">= 0.6.0 < 0.11.0"} (current: {"node":"5.1.0","npm":"3.4.1"})
Installing apm...
module.js:340
    throw err;
    ^

**Comment by steezeburger** [comment]
@middle-aged-man I'm also on Arch. I would recommend using nvm. It's easy to use and has helped me a lot. NPM has been kind of quirky lately, and using nvm has helped reduce the headaches.

**Comment by conraid** [comment]
@InsidiousMind no, npm 3.5.0
https://github.com/npm/npm/releases/tag/v3.5.0

**Comment by Narrat** [comment]
Upstream (npm) released 3.5.x, the issue itself was only with 3.4.1, the mentioned distribution is already at the latest version, the issue itself got something to do with npm, so in my opinion this issue could be closed

**Comment by 50Wliu** [comment]
Ok, thanks @Narrat!
